src/domain/users/user-organizations.repository.ts

```diff
@@ -0,0 +1,305 @@
+        status: args.status,
+      });
+
+      const isActivatingUserOrg = args.status === UserOrganizationStatus.ACTIVE;
+      const isUserPending = signer.user.status === UserStatus.PENDING;
+      if (isActivatingUserOrg && isUserPending) {
+        await entityManager.update(DbUser, signer.user.id, {
+          status: UserStatus.ACTIVE,
+        });
+      }
+    });
+  }
+
+  public async get(args: {
+    authPayload: AuthPayload;
+    orgId: Organization['id'];
+  }): Promise<Array<UserOrganization>> {
+    const signer = await this.getOrgAndSignerOrFail(args);
+    return signer.org.user_organizations;
+  }
+
+  public async updateRole(args: {
+    authPayload: AuthPayload;
+    orgId: Organization['id'];
+    userId: User['id'];
+    role: UserOrganization['role'];
+  }): Promise<void> {
+    const signer = await this.getOrgAndSignerOrFail(args);
+
+    this.assertUserOrgIsActive(signer.userOrg);
+    this.assertUserOrgAdmin(signer.userOrg);
+    if (args.role !== UserOrganizationRole.ADMIN) {
+      this.assertNotLastActiveAdmin(signer.org.user_organizations, args.userId);
+    }
+
+    const updateUserOrg = this.findUserOrg({
+      userOrgs: signer.org.user_organizations,
+      userId: args.userId,
+    });
+
+    const userOrganizationRepository =
+      await this.postgresDatabaseService.getRepository(DbUserOrganization);
+    await userOrganizationRepository.update(
+      { id: updateUserOrg.id },
+      { role: args.role },
+    );
+  }
+
+  public async removeUser(args: {
+    authPayload: AuthPayload;
+    userId: User['id'];
+    orgId: Organization['id'];
+  }): Promise<void> {
+    const signer = await this.getOrgAndSignerOrFail(args);
+
+    this.assertUserOrgIsActive(signer.userOrg);
+    this.assertUserOrgAdmin(signer.userOrg);
+    this.assertNotLastActiveAdmin(signer.org.user_organizations, args.userId);
+
+    const updateUserOrg = this.findUserOrg({
+      userOrgs: signer.org.user_organizations,
+      userId: args.userId,
+    });
+
+    const userOrganizationRepository =
+      await this.postgresDatabaseService.getRepository(DbUserOrganization);
+    await userOrganizationRepository.delete(updateUserOrg.id);
+  }
+
+  private async getOrgAndSignerOrFail(args: {
+    authPayload: AuthPayload;
+    orgId: Organization['id'];
+  }): Promise<{
+    org: Organization;
+    user: User;
+    userOrg: UserOrganization;
+  }> {
+    this.assertSignerAddress(args.authPayload);
+
+    const user = await this.usersRepository.findByWalletAddressOrFail(
+      args.authPayload.signer_address,
+    );
+    const org = await this.organizationsRepository.findOneOrFail({
+      where: { id: args.orgId },
+      relations: { user_organizations: { user: true } },
+    });
+    const userOrg = this.findUserOrg({
+      userOrgs: org.user_organizations,
+      userId: user.id,
+    });
+
+    return { org, user, userOrg };
+  }
+
+  private findUserOrg(args: {
```

Comment:
We should rename this as `find*` is confusing. It appears as though it's a database operation.